scripts/firedrake-install

```diff
@@ -1160,21 +1172,20 @@ if mode == "install":
 
     # Need to install petsc first in order to resolve hdf5 dependency.
     if not args.honour_petsc_dir:
-        pipinstall.append("--no-deps")
-        packages.remove("petsc")
-        install("petsc/")
-        pipinstall.pop()
+        with pipargs("--no-deps"):
+            packages.remove("petsc")
+            install("petsc/")
 
     for p in packages:
         pip_requirements(p)
 
     build_and_install_h5py()
     build_and_install_libspatialindex()
     build_and_install_libsupermesh()
-    pipinstall.append("--no-deps")
-    for p in packages:
-        install(p+"/")
-        sys.path.append(os.getcwd() + "/" + p)
+    with pipargs("--no-deps"):
+        for p in packages:
+            install(p+"/")
```

Comment:
Sympy now requires `mpmath` as a separate package. We install sympy when installing FIAT, but with `--no-deps`, this breaks the sympy installation (hence the present build error).

Reply:
Did this just happen? I don't understand why this change changes anything.

Reply:
Not just now, I think (sympy >= 1.0, we're presently on 1.3). Like you, I don't understand why this change would break things in this way.

Reply:
So what's the fix?

Reply:
No clue (I got as far as noticing why the build had failed).

Reply:
OK I'll come back to this later.